protoc_plugin/lib/src/protobuf_field.dart

```diff
@@ -63,9 +64,32 @@ class ProtobufField {
   bool get isRepeated =>
       descriptor.label == FieldDescriptorProto_Label.LABEL_REPEATED;
 
-  /// True if the field is to be encoded with [packed=true] encoding.
-  bool get isPacked =>
-      isRepeated && descriptor.options != null && descriptor.options.packed;
+  /// True if a numeric field is repeated and it should be encoded with packed
+  /// encoding. In proto3 repeated fields are encoded as packed by default.
+  /// proto2 requires `[packed=true]` option.
+  bool get isPacked {
+    if (!isRepeated) {
+      return false; // only repeated fields can be packed
```

Comment:
Should this instead throw an argument-error (is it not an error to ask if a non-repeated field is packed?)

Reply:
Looking at the call-site it makes sense to just return false here I think.

I would prefer not inlining this amount of logic.